### ISSUE TITLE
Failed: akka.http.scaladsl.server.directives.FileUploadDirectivesSpec.the fileUploadAll directive should stream all file uploads if multiple with the same name are posted as streamed entities

### ISSUE THREAD

**Issue by jrudolph** [akka/akka-http#2733]
https://jenkins.akka.io:8498/job/akka-http-nightly/639/AKKA_VERSION=release-2.5,Node=akka-http-nightly-node,SCALA_VERSION=2.12.9,jdk=AdoptOpenJDK%208/testReport/junit/akka.http.scaladsl.server.directives/FileUploadDirectivesSpec/the_fileUploadAll_directive_should_stream_all_file_uploads_if_multiple_with_the_same_name_are_posted_as_streamed_entities/

```
Stacktrace

org.scalatest.exceptions.TestFailedException: Request was neither completed nor rejected within 3 seconds
	at akka.http.scaladsl.testkit.TestFrameworkInterface$Scalatest.failTest(TestFrameworkInterface.scala:24)
	at akka.http.scaladsl.testkit.TestFrameworkInterface$Scalatest.failTest$(TestFrameworkInterface.scala:24)
	at akka.http.scaladsl.server.RoutingSpec.failTest(RoutingSpec.scala:19)
	at akka.http.scaladsl.testkit.RouteTestResultComponent$RouteTestResult.failNeitherCompletedNorRejected(RouteTestResultComponent.scala:93)
	at akka.http.scaladsl.testkit.RouteTestResultComponent$RouteTestResult.rawResponse(RouteTestResultComponent.scala:58)
	at akka.http.scaladsl.testkit.RouteTestResultComponent$RouteTestResult.response(RouteTestResultComponent.scala:39)
	at akka.http.scaladsl.testkit.RouteTest.response(RouteTest.scala:61)
	at akka.http.scaladsl.testkit.RouteTest.response$(RouteTest.scala:61)
	at akka.http.scaladsl.server.RoutingSpec.response(RoutingSpec.scala:19)
	at akka.http.scaladsl.testkit.RouteTest.status(RouteTest.scala:79)
	at akka.http.scaladsl.testkit.RouteTest.status$(RouteTest.scala:79)
	at akka.http.scaladsl.server.RoutingSpec.status(RoutingSpec.scala:19)
	at akka.http.scaladsl.server.directives.FileUploadDirectivesSpec.$anonfun$new$70(FileUploadDirectivesSpec.scala:407)
	at scala.util.DynamicVariable.withValue(DynamicVariable.scala:62)
	at akka.http.scaladsl.testkit.RouteTest.$anonfun$check$1(RouteTest.scala:56)
	at akka.http.scaladsl.testkit.RouteTestResultComponent$RouteTestResult.$tilde$greater(RouteTestResultComponent.scala:50)
	at akka.http.scaladsl.server.directives.FileUploadDirectivesSpec.$anonfun$new$69(FileUploadDirectivesSpec.scala:406)
```

**Comment by vaslabs** [comment]
I have the same issue, my test fails on a 50-50 basis.
Always failing on the file upload part.
The route definition looks like this
```
path("upload") {
    post {
        uploadFile ~ entity(as[JsonSomething]) { json => ...}
    }
}
```
This also happens on live mode as well at the same 50-50 rate for me and always for the upload file

**Comment by vaslabs** [comment]
Ok I think that was my bad, I did some changes and now I get higher success rates.
For anyone reading in the future:
What I did before:
- Accepting two form data bodies, one large binary file and one json
-    The json was short so I was materialising and consuming the stream
-    I wanted to give the status code based on json parsing
-    This was giving me 50% success rate

What I did after:

-     I just build the two Source definitions, zip them together and passing them to the API
-     The status now it's just an accepted, the connection remains open and when the upload finishes it streams out a json body indicating success/error

**Comment by johanandren** [comment]
https://jenkins.akka.io:8498/job/akka-http-nightly/AKKA_VERSION=default,Node=akka-http-nightly-node,SCALA_VERSION=2.11.12,jdk=AdoptOpenJDK%208/749/

**Comment by chbatey** [comment]
https://jenkins.akka.io:8498/job/akka-http-nightly/AKKA_VERSION=default,Node=akka-http-nightly-node,SCALA_VERSION=2.12.10,jdk=AdoptOpenJDK%2011/81/testReport/junit/akka.http.scaladsl.server.directives/FileUploadDirectivesSpec/the_fileUploadAll_directive_should_stream_all_file_uploads_if_multiple_with_the_same_name_are_posted_as_strict_entities/

**Comment by chbatey** [comment]
https://jenkins.akka.io:8498/job/akka-http-nightly/AKKA_VERSION=master,Node=akka-http-nightly-node,SCALA_VERSION=2.13.1,jdk=AdoptOpenJDK%208/143/testReport/junit/akka.http.scaladsl.server.directives/FileUploadDirectivesSpec/the_fileUploadAll_directive_should_stream_all_file_uploads_if_multiple_with_the_same_name_are_posted_as_streamed_entities/

**Comment by jrudolph** [comment]
Hopefully fixed this time by #3206.
